cellprofiler/modules/tophattransform.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+
+"""
+
+<strong>Top-hat transform</strong>
+
+"""
+
+import cellprofiler.image
+import cellprofiler.module
+import cellprofiler.setting
+import skimage.morphology
+
+
+class TopHatTransform(cellprofiler.module.Module):
+    category = "Mathematical morphology"
```

Comment:
The mathematical morphology modules are all very similar. Perhaps we could create a `cellprofiler.module.MathematicalMorthpholoy` class which encompasses the repeated behavior.